server/src/db-client.ts

```diff
@@ -0,0 +1,20 @@
+import Knex from "knex";
+import type { Knex as KnexType } from "knex";
+import { logger } from "./logger";
+import { getHostInstance, loadDbConfig } from "./config";
+
+export function createDbClient(label: string): KnexType<any, unknown[]> {
+  const client = Knex(loadDbConfig());
+
+  // Add debug-related logging.
+  const pool = client.client.pool;
+  const instanceName = getHostInstance();
+  function logPoolSize() {
+    const poolSize = pool.numUsed() + pool.numFree();
+    logger.info(`${instanceName} ${label} DB pool size: ${poolSize}`);
```

Comment:
Once we have a handle on this, I‘m going to flip it to `logger.debug`. But for now, actually printing debug logs is a lot noisier than is worthwhile.